images/virtualization-artifact/pkg/controller/k8s-validation/validate-k8s-utils.go

```diff
@@ -0,0 +1,337 @@
+/*
+Copyright 2014 The Kubernetes Authors.
+Copyright 2024 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+
+The code here is taken from https://github.com/kubernetes/kubernetes/blob/29fb8e8b5a41b2a7d760190284bae7f2829312d3/pkg/apis/core/validation/validation.go#L3288
+The "core" package is change to exported package "k8s.io/api/core/v1" in
+order to avoid dependency on kubernetes/kubernetes
+
+https://github.com/kubernetes/kubernetes/blame/29fb8e8b5a41b2a7d760190284bae7f2829312d3/pkg/apis/core/validation/validation.go#L3288
+the code hardly changes all of the changes have been atleast a few years older
+this makes it easier to copy and maintain instead of vendoring in kubernetes or
+creating dry runs of the pod object during admission validation.
```

Comment:
```suggestion
The code is very stable, it was barely changed during the past years.
It makes it easier to copy and maintain instead of vendoring the whole kubernetes as module or
creating dry runs of the pod object during admission validation.
```